bugsnag-android-core/src/main/java/com/bugsnag/android/DeliveryStyle.java

```diff
@@ -1,8 +1,6 @@
 package com.bugsnag.android;
 
 enum DeliveryStyle {
-    SAME_THREAD,
```

Comment:
What purpose did `SAME_THREAD` delivery style / `notifyBlocking()` method serve and why is it ok that it hasn't been replaced?

Reply:
`notifyBlocking()` generated and then delivered a `Report` on the same thread. In contrast, we now always generate an `Event`, then deliver it on a background thread.

Blocking delivery is not what the vast majority of users would want as this is likely to cause ANRs in a production app, by blocking the main thread. `notifyBlocking()` also exposes additional complexity to the user and forces them to make a conscious choice between `notify()` and `notifyBlocking()` - when the first option is the right one. Additionally this is no longer in the notifier spec so removing this method is more consistent.

Removing `notifyBlocking()` means that the enum only has two options. The `DeliveryStyle` really wants to check whether a report is unhandled and cache it to disk before attempting delivery if that is the case. Therefore the `Client` has been updated to do this and `DeliveryStyle` has been removed as it is obsolete.